libparsec/crates/platform_mountpoint/tests/base.rs

```diff
@@ -0,0 +1,207 @@
+    let (path, fs) = mount!(tmp_path);
+
+    // TODO: assert_eq it
+    // ErrorKind::InvalidFilename is under `io_error_more` feature which is
+    // available on nightly only
+    std::fs::create_dir(path.join(name)).unwrap_err();
+
+    fs.stop();
+}
+
+#[parsec_test]
+#[case("x".repeat(256))]
+#[case("飞".repeat(85) + "x")]
+fn file_exceeding_255_bytes(tmp_path: TmpPath, #[case] name: String) {
+    let (path, fs) = mount!(tmp_path);
+
+    // TODO: assert_eq it
+    // ErrorKind::InvalidFilename is under `io_error_more` feature which is
+    // available on nightly only
+    File::create(path.join(name)).unwrap_err();
+
+    fs.stop();
+}
+
+#[parsec_test]
+fn read_write(tmp_path: TmpPath) {
+    let (path, fs) = mount!(tmp_path);
+    let bar = path.join("bar");
+    let data = b"Lorem ipsum";
+
+    File::create(&bar).unwrap();
+
+    std::fs::write(&bar, data).unwrap();
+
+    p_assert_eq!(std::fs::read(&bar).unwrap(), data);
+
+    fs.stop();
+}
+
+#[parsec_test]
+fn remove_dir(tmp_path: TmpPath) {
+    let (path, fs) = mount!(tmp_path);
+    let bar = path.join("bar");
+
+    std::fs::create_dir(&bar).unwrap();
+    std::fs::remove_dir(&bar).unwrap();
+
+    fs.stop();
+}
+
+#[parsec_test]
+fn remove_file(tmp_path: TmpPath) {
+    let (path, fs) = mount!(tmp_path);
+    let bar = path.join("bar");
+
+    File::create(&bar).unwrap();
+    std::fs::remove_file(&bar).unwrap();
+
+    fs.stop();
+}
+
+#[parsec_test]
+fn rename_dir(tmp_path: TmpPath) {
+    let (path, fs) = mount!(tmp_path);
+    let bar = path.join("bar");
+    let foo = path.join("foo");
+
+    std::fs::create_dir(&bar).unwrap();
+    std::fs::rename(&bar, foo).unwrap();
+
+    fs.stop();
+}
+
+#[parsec_test]
+fn rename_file(tmp_path: TmpPath) {
+    let (path, fs) = mount!(tmp_path);
+    let bar = path.join("bar");
+    let foo = path.join("foo");
+
+    File::create(&bar).unwrap();
+    std::fs::rename(&bar, foo).unwrap();
+
+    fs.stop();
+}
+
+#[parsec_test]
+fn rename_dir_non_empty(tmp_path: TmpPath) {
+    let (path, fs) = mount!(tmp_path);
+    let bar = path.join("bar");
+    let foo = path.join("foo");
+    let baz = bar.join("baz");
+    let new_baz = foo.join("baz");
+
+    std::fs::create_dir(&bar).unwrap();
+    File::create(baz).unwrap();
+    std::fs::rename(&bar, &foo).unwrap();
+
+    assert!(new_baz.exists());
+
+    fs.stop();
+}
+
+#[parsec_test]
+fn copy(tmp_path: TmpPath) {
+    let (path, fs) = mount!(tmp_path);
+    let bar = path.join("bar");
+    let foo = path.join("foo");
+    let data = b"Lorem ipsum";
+
+    File::create(&bar).unwrap();
+    File::create(&foo).unwrap();
+    println!("{} {}", bar.exists(), foo.exists());
+    std::fs::write(&bar, data).unwrap();
+
+    std::fs::copy(&bar, &foo).unwrap();
+
+    p_assert_eq!(std::fs::read(&bar).unwrap(), data);
+    p_assert_eq!(std::fs::read(&foo).unwrap(), data);
+
+    fs.stop();
+}
+
+#[parsec_test]
+fn delete_then_close_file(tmp_path: TmpPath) {
```

Comment:
Previous tests are named "remove"
```suggestion
fn remove_then_close_file(tmp_path: TmpPath) {
```